README.md

```diff
@@ -1,32 +1,399 @@
-# Snowplow Ruby tracker example
+# Snowplow Ruby tracker examples
 
 An example of how to incorporate [Snowplow](https://snowplowanalytics.com/) trackers (SDKs) into a Rails project.
 
-### Quick Start
+### Versions used:
 
-The [Ruby](https://github.com/snowplow/snowplow-ruby-tracker) and [JavaScript](https://github.com/snowplow/snowplow-javascript-tracker) trackers are configured to use [Snowplow Micro](https://github.com/snowplow-incubator/snowplow-micro) as the event collector. The Micro config files are included in the `snowplow-micro` folder.
+**Rails** v6.1.4  
+**Ruby tracker** v0.6.1  
+**JavaScript tracker** v3.1.0
+
+## Table of Contents
+
+- [Snowplow Ruby tracker examples](#snowplow-ruby-tracker-examples)
+    - [Versions used:](#versions-used)
+  - [Table of Contents](#table-of-contents)
+  - [1. Quick Start](#1-quick-start)
+  - [2. Tracking Design and Implementation](#2-tracking-design-and-implementation)
+    - [2.1 This demo app](#21-this-demo-app)
+    - [2.2 Ruby tracker](#22-ruby-tracker)
+    - [2.3 JavaScript tracker](#23-javascript-tracker)
+  - [3. Event types and context](#3-event-types-and-context)
+  - [4. Matching the domain_userid for both trackers](#4-matching-the-domain_userid-for-both-trackers)
+  - [5. Testing using Snowplow Micro](#5-testing-using-snowplow-micro)
+  - [6. Further information](#6-further-information)
+
+## 1. Quick Start
+
+**Requirements**: Ruby 3.0, Yarn, Bundle, Docker, and Cypress (for the tests).
+
+Install project dependencies:
+
+```bash
+yarn install
+bundle install
+```
+
+Run the app:
+
+```bash
+# To run the app on port 3000
+rails server
+```
+
+The [Ruby](https://github.com/snowplow/snowplow-ruby-tracker) and [JavaScript](https://github.com/snowplow/snowplow-javascript-tracker) trackers are configured to use [Snowplow Micro](https://github.com/snowplow-incubator/snowplow-micro) as the event collector.
 
 Start Micro:
 
+```bash
+# run from the root folder of this app
+
+docker run --mount type=bind,source=$(pwd)/snowplow-micro,destination=/config -p 9090:9090 snowplow/snowplow-micro:1.1.2 --collector-config /config/micro.conf --iglu /config/iglu.json
+```
+
+Interact with the site to generate events.
+
+Run tests:
+
+```bash
+# Rails tests
+rspec
+
+# Start the Rails server on port 5017 for Cypress
+bundle exec rails server -e test -p 5017
+
+# Opens the Cypress UI for headed testing
+yarn cypress open --project ./spec
```

Comment:
I think it'd be cool to wrap the `bundle` command into a single `package.json` command you can run with yarn.

This is a nice example of all the fun you can have with package.json scripts: https://github.com/snowplow-incubator/snowplow-react-native-tracker/blob/master/DemoApp/package.json#L5-L27